capstone/deployment.py

```diff
@@ -101,3 +111,137 @@ def add_sys_path(path):
 
 def get_deployment_root() -> Path:
     return Path(config.data_dir) / "deployments"
+
+NOMAD_JOB_TEMPLATE = """
+job "{{name}}" {
+  type = "service"
+
+  group "{{name}}" {
+    count = 1
+
+    network {
+      port "web" {
+        to = 8080
+      }
+    }
+
+    service {
+      name     = "{{name}}"
+      port     = "web"
+
+      tags = ["capstone-service"]
+
+      meta {
+        host = "{{host}}"
+      }
+    }
+
+    task "{{name}}" {
+
+      driver = "docker"
+
+      config {
+        image = "{{docker_image}}"
+        force_pull = true
+        ports = ["web"]
+      }
+    }
+  }
+}
+"""
+
+def get_nomad_job_hcl(name, host, docker_image):
+    t = Template(NOMAD_JOB_TEMPLATE)
+    return t.render(name=name, host=host, docker_image=docker_image)
+
+class Task:
+    def __init__(self, site: str):
+        self.site = site
+        self.task_id = uuid.uuid4().hex
+        self.logger = self.make_logger(self.site, self.task_id)
+
+    def make_logger(self, site: str, task_id: str) -> logging.Logger:
+        '''Creates a logger for a particular upload and sets up relevant parameters
+        '''
+        log_root = Path(config.data_dir)
+
+        logger = logging.Logger(task_id)
+        logger.setLevel(logging.DEBUG)
+        filename = f'{task_id}.log'
+        logfile = log_root.joinpath("tasks", site, filename)
+        logfile.parent.mkdir(parents=True, exist_ok=True)
+        handler = logging.FileHandler(str(logfile))
+        formatter = logging.Formatter(
+            fmt=f'[%(asctime)s] [%(levelname)s] %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S'
+        )
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        logger.addHandler(logging.StreamHandler(sys.stdout))
```

Comment:
Why is this logging to stdout and not stderr?

Reply:
no real preference. Using stderr was messing up with some earlier prints, so started using stdout. I'm okay to move to stderr.